HybridCar.cpp

```diff
@@ -1,19 +1,29 @@
 #include "HybridCar.hpp"
 #include <iostream>
 
-HybridCar::HybridCar(PetrolEngine* petrolEng, ElectricEngine* electricEng)
-    : petrolEngine_(petrolEng)
-    , electricEngine_(electricEng)
-{
+HybridCar::HybridCar(std::unique_ptr<PetrolEngine> petrolEng, std::unique_ptr<ElectricEngine> electricEng)
+    : PetrolCar(std::move(petrolEng)), ElectricCar(std::move(electricEng)) {
     std::cout << __FUNCTION__ << std::endl;
 }
 
-HybridCar::~HybridCar()       { std::cout << __FUNCTION__ << std::endl; }
-void HybridCar::turnLeft()      { std::cout << __FUNCTION__ << std::endl; }
-void HybridCar::turnRight()     { std::cout << __FUNCTION__ << std::endl; }
-void HybridCar::brake()         { std::cout << __FUNCTION__ << std::endl; }
-void HybridCar::accelerate(int) { std::cout << __FUNCTION__ << std::endl; }
-void HybridCar::charge()        { std::cout << __FUNCTION__ << std::endl; }
-void HybridCar::refuel()        { std::cout << __FUNCTION__ << std::endl; }
- 
+HybridCar::~HybridCar() {
+    std::cout << __FUNCTION__ << std::endl;
+}
+
+void HybridCar::restore() {
+    PetrolCar::restore();
+    ElectricCar::restore();
+}
+
+void HybridCar::printInfoEngine() const {
+    PetrolCar::printInfoEngine();
+    ElectricCar::printInfoEngine();
+}
+
+void HybridCar::changePetrolEngine(std::unique_ptr<PetrolEngine> petrolEngine) {
+    PetrolCar::changeEngine(std::move(petrolEngine));
+}
 
+void HybridCar::changeElectricEngine(std::unique_ptr<ElectricEngine> electricEngine) {
+    ElectricCar::changeEngine(std::move(electricEngine));
```

Comment:
redundant functions. We inherit changeEngine from base classes, so we can call changeEngine with either unique_ptr to Petrol or Electric engine.